symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV1.java

```diff
@@ -0,0 +1,138 @@
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Slf4j
+public class DatafeedServiceV1 extends AbstractDatafeedService {
+
+    private static final String DATAFEED_ID_FILE = "datafeed.id";
+    private final AtomicBoolean started = new AtomicBoolean();
+    private String datafeedId;
+
+
+    public DatafeedServiceV1(ApiClient agentClient, ApiClient podClient, AuthSession authSession, BdkConfig config) {
+        super(agentClient, podClient, authSession, config);
+        this.started.set(false);
+        this.datafeedId = null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() throws Throwable {
+        this.datafeedId = this.retrieveDatafeedIdFromDisk();
+
+        if (this.datafeedId == null) {
+            this.datafeedId = this.createDatafeedAndSaveToDisk();
+        }
+
+        log.debug("Start reading events from datafeed {}", datafeedId);
+        do {
+            this.started.set(true);
+            this.readDatafeed();
+        } while (this.started.get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stop() {
+        this.started.set(false);
+        this.datafeedId = null;
+    }
+
+    private void readDatafeed() throws Throwable {
+        this.retry.executeCheckedSupplier(() -> {
+            try {
+                List<V4Event> events = datafeedApi.v4DatafeedIdReadGet(datafeedId, authSession.getSessionToken(), authSession.getKeyManagerToken(), null);
+                if (events != null && !events.isEmpty()) {
+                    handleV4EventList(events);
+                }
+            } catch (ApiException e) {
+                log.error("Error {}: {}", e.getCode(), e.getMessage());
+                if (e.isUnauthorized()) {
+                    log.info("Re-authenticate and try again");
+                    authSession.refresh();
+                }
+                if (e.isClientError()) {
+                    log.info("Recreate a new datafeed and try again");
+                    datafeedId = this.createDatafeedAndSaveToDisk();
+                }
+                throw e;
+            }
+            return null;
+        });
+    }
+
+    private String createDatafeedAndSaveToDisk() throws Throwable {
+        log.debug("Start creating a new datafeed and save to disk");
+        return this.retry.executeCheckedSupplier(() -> {
+            try {
+                String id = this.datafeedApi.v4DatafeedCreatePost(authSession.getSessionToken(), authSession.getKeyManagerToken()).getId();
+                this.writeDatafeedIdToDisk(id);
+                log.debug("Datafeed: {} was created and saved to disk", id);
+                return id;
+            } catch (ApiException e) {
+                log.error("Error {}: {}", e.getCode(), e.getMessage());
+                if (e.isUnauthorized()) {
+                    log.info("Re-authenticate and try again");
+                    authSession.refresh();
+                }
+                throw e;
```

Comment:
so if auth has expired it means we still throw the exception and go into the retry from resilience4j and wait before retrying the request where we could retry directly